library_information_validator.py

```diff
@@ -57,12 +59,11 @@ def projectID(self):
         if(len(re.findall('P\d+P\d+', plate_id))>0):
             project_id_user = re.findall('P\d+', plate_id)[0]
         else:
-            logger.error(
+            sys.exit(logger.error(
```

Comment:
The argument for `sys.exit()` is usually a number which corresponds to an exit code. `0` is good, anything other than `0` is bad (error) - typically `1` when writing code.

`logger.error` is probably returning `None` or something else odd. What you want instead I think is the `logger.error` call followed by `sys.exit(1)`...

Reply:
Actually, I got the idea to do it like this from the python documentation. There is says as follows:
"The optional argument arg can be an integer giving the exit status (defaulting to zero), or another type of object. [...]  If another type of object is passed, None is equivalent to passing zero, and any other object is printed to stderr and results in an exit code of 1. In particular, sys.exit("some error message") is a quick way to exit a program when an error occurs."

Reply:
Right - but this doesn't really change my point. `logger.error` always returns `None` (I just checked), which means that you are always exiting with exit status `0` - this is interpreted as exiting successfully.

Reply:
ps. To clarify: `sys.exit("some error message")` would exit with a non-zero exit code. It's just `sys.exit(logger.error("some error message"))` which won't..

Reply:
:( and I thought I was right for once. But I bow to the master and will change it.

Reply:
![beard](https://media.giphy.com/media/13pdBSoCcDejrG/giphy.gif)